packages/pagination/src/hooks/usePagination.ts

```diff
@@ -0,0 +1,117 @@
+
+const usePagination = ({
+  onClick,
+  currentPage,
+  totalPages,
+}: UsePaginationProps) => {
+  const { breakpoints } = useBreakpoint();
+  const isDesktop = breakpoints.isDesktop;
+
+  const shouldShowPreDots =
+    (currentPage > 3 && totalPages !== 4) || currentPage > 4;
+  const shouldShowPostDots =
+    (currentPage < totalPages - 2 && totalPages !== 4) ||
+    currentPage < totalPages - 3;
+
+  const siblingsToRender = getSiblingsToRender(totalPages, currentPage);
+
+  const items: PaginationItemProps[] = isDesktop
+    ? [
+        // Previous button
+        {
+          type: paginationItemType.previous,
+          page: currentPage - 1,
+          isDisabled: currentPage === 1,
+          onClick: parseOnClick(onClick, currentPage - 1, currentPage === 1),
+        },
+        // First page
+        {
+          type: paginationItemType.page,
+          page: 1,
+          isCurrentPage: currentPage === 1,
+          onClick: parseOnClick(onClick, 1, false),
+        },
+        // Ellipsis
+        ...shouldReturnObj(shouldShowPreDots, {
+          type: paginationItemType.ellipsis,
+        }),
+        // Siblings
+        ...siblingsToRender.map(page => ({
+          type: paginationItemType.page,
+          page: page,
+          isCurrentPage: currentPage === page,
+          onClick: parseOnClick(onClick, page, false),
+        })),
+        // Ellipsis
+        ...shouldReturnObj(shouldShowPostDots, {
+          type: paginationItemType.ellipsis,
+        }),
+        // Last Page
+        ...shouldReturnObj(totalPages > 1, {
+          type: paginationItemType.page,
+          page: totalPages,
+          isCurrentPage: currentPage === totalPages,
+          onClick: parseOnClick(onClick, totalPages, false),
+        }),
+        // Next button
+        {
+          type: paginationItemType.next,
+          page: currentPage + 1,
+          isDisabled: currentPage === totalPages,
+          onClick: parseOnClick(
+            onClick,
+            currentPage + 1,
+            currentPage === totalPages,
+          ),
+        },
+      ]
+    : [
+        // Previous button
+        {
+          type: paginationItemType.previous,
+          page: currentPage - 1,
+          isDisabled: currentPage === 1,
+          onClick: parseOnClick(onClick, currentPage - 1, currentPage === 1),
+        },
+        // Current page
+        ...shouldReturnObj(currentPage !== totalPages || totalPages === 1, {
+          type: paginationItemType.page,
+          page: currentPage,
+          onClick: parseOnClick(onClick, currentPage, false),
+        }),
+        // Slash
+        ...shouldReturnObj(currentPage !== totalPages, {
+          type: paginationItemType.slash,
+        }),
+        // Last Page
+        ...shouldReturnObj(totalPages > 1, {
+          type: paginationItemType.page,
+          page: totalPages,
+          onClick: parseOnClick(onClick, totalPages, false),
```

Comment:
are we not checking `isCurrentPage` on this one [as we do on Desktop](https://github.com/hey-car/heycar-uikit/pull/123/files#diff-f36f31157bfa36f852170b170d5df109528e600230bf058642233bf90a8f0079R62)?

Reply:
We don't have this check because in mobile, the last page is always selectable and no page has a 'isCurrent' status, this is in compliance with design guidelines.

Reply:
@frankie303 First and last I was not able to merge because of the comment above, but previous and next buttons yes.